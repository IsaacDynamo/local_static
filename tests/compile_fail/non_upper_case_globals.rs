#![deny(warnings)]

use local_static::local_static;

#[local_static]
fn main() {
    static mut snake_case: u32 = 0;
    *snake_case += 1;
}
