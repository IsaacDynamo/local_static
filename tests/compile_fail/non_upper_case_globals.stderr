error: static variable `snake_case` should have an upper case name
 --> tests/compile_fail/non_upper_case_globals.rs:7:16
  |
7 |     static mut snake_case: u32 = 0;
  |                ^^^^^^^^^^ help: convert the identifier to upper case: `SNAKE_CASE`
  |
note: the lint level is defined here
 --> tests/compile_fail/non_upper_case_globals.rs:1:9
  |
1 | #![deny(warnings)]
  |         ^^^^^^^^
  = note: `#[deny(non_upper_case_globals)]` implied by `#[deny(warnings)]`
